File: Java/day15.java
import java.util.*;

class Node
{
	int data;
	Node next;
	Node(int d) {
        data = d;
        next = null;
    }
}

// all instances of "day15" below renamed from Solution for easier compilation
public class day15
{
    public static Node insert(Node head, int data) {
        // Complete this method
        if (head == null)
            return new Node(data);
        else {
            Node current = head;
            while (current.next != null)
                current = current.next;
            Node new_node = new Node(data);
            current.next = new_node;
            return head;
        }
	}

    public static void display(Node head) {
        Node start = head;
        while(start != null) {
            System.out.print(start.data + " ");
            start = start.next;
        }
    }

    public static void main(String args[]) {
        Scanner sc = new Scanner(System.in);
        Node head = null;
        int T = sc.nextInt();
        while (T-- > 0) {
            int ele = sc.nextInt();
            head = insert(head, ele);
        }
        display(head);
    }
}

File: Scala/day15.scala
class Node(d: Int) {
  var data: Int = d
  var next: Node = null
}

class NodeLinkedList {
  def display(head: Node): Unit = {
    var start: Node = head
    while (start != null) {
      print(start.data + " ")
      start = start.next
    }
  }

  def insert(head: Node, data: Int): Node = { // complete this method
    if (head == null) {
      val new_head: Node = new Node(data)
      return new_head
    }
    else {
      var current: Node = head
      while (current.next != null) current = current.next
      val new_node: Node = new Node(data)
      current.next = new_node
      return head
    }
  }
}

object Solution {
  def main(args: Array[String]) {
    val sc = new java.util.Scanner(System.in)
    val list: NodeLinkedList = new NodeLinkedList
    var head: Node = null
    var T: Int = sc.nextInt()
    for (i <- 1 to T) {
      var e = sc.nextInt()
      head = list.insert(head, e)
    }

    list.display(head)
  }
}
